src/js/component.js

```diff
@@ -1283,6 +1287,94 @@ class Component {
     return this.currentDimension('height');
   }
 
+  /**
+   * Retrieves the position and size information of the component's element.
+   *
+   * @return {Object} An object with `boundingClientRect` and `center` properties.
+   *         - `boundingClientRect`: An object with properties `x`, `y`, `width`,
+   *           `height`, `top`, `right`, `bottom`, and `left`, representing
+   *           the bounding rectangle of the element.
+   *         - `center`: An object with properties `x` and `y`, representing
+   *           the center point of the element. `width` and `height` are set to 0.
+   */
+  getPositions() {
+    const rect = this.el_.getBoundingClientRect();
+
+    // Creating objects that mirror DOMRectReadOnly for boundingClientRect and center
+    const boundingClientRect = {
+      x: rect.x,
+      y: rect.y,
+      width: rect.width,
+      height: rect.height,
+      top: rect.top,
+      right: rect.right,
+      bottom: rect.bottom,
+      left: rect.left
+    };
+
+    // Calculating the center position
+    const center = {
+      x: rect.left + rect.width / 2,
+      y: rect.top + rect.height / 2,
+      width: 0,
+      height: 0,
+      top: rect.top + rect.height / 2,
+      right: rect.left + rect.width / 2,
+      bottom: rect.top + rect.height / 2,
+      left: rect.left + rect.width / 2
+    };
+
+    return {
+      boundingClientRect,
+      center
+    };
+  }
+
+  /**
+   * Handles the focus event. This method could be overridden
+   * in subclasses to provide specific focus event handling.
+   *
+   * Calls for handling of the Player Focus if:
+   * * SpatialNavigation is enabled, not paused & element is focusable.
+   */
+  handleFocus() {
+    const spatialNavigation = this.player_.spatialNavigation;
+
+    if (spatialNavigation && this.getIsFocusable(this.el_)) {
+      spatialNavigation.handlePlayerFocus();
+    }
+  }
+
+  /**
+   * Handles the blur event. This method could be overridden
+   * in subclasses to provide specific focus event handling.
+   *
+   * @param {string|Event|Object} event
+   *        The name of the event, an `Event`, or an object with a key of type set to
+   *        an event name.
+   *
+   * Calls for handling of the Player Blur if:
+   * * Next focused element is not a children of current focused element &
+   *   next focused element is not a children of Player.
+   * * There is no next focused element
+   */
+  handleBlur(event) {
+    const spatialNavigation = this.player_.spatialNavigation;
+
+    if (spatialNavigation && this.getIsFocusable(this.el_)) {
+      const nextFocusedElement = event.relatedTarget;
+      let isChildrenOfPlayer = null;
+
+      if (nextFocusedElement) {
+        isChildrenOfPlayer = Boolean(nextFocusedElement.closest('.video-js'));
+      }
+
+      if (!(event.currentTarget.contains(event.relatedTarget)) && !isChildrenOfPlayer || !nextFocusedElement) {
+        spatialNavigation.handlePlayerBlur(this);
```

Comment:
On that note, maybe we can have a check before any of this logic is executed that checks if the component is a player

Or even better, don't bind these spatial navigation specific functions if the component is the player

Reply:
I don't quite understand the purpose of this function, because even if it's removed it seems to work correctly. What would be the use case where it's needed?

Same applies to `handleFocus`.

Reply:
> I don't quite understand the purpose of this function, because even if it's removed it seems to work correctly. What would be the use case where it's needed?
> 
> Same applies to `handleFocus`.

Hello @amtins ,  A use for  case 'handleBlur' would be: The user presses down when is focus is in the control bar, selects a component below the player & then presses up to regain focus of the last selected component inside the control bar. This is possible because we conditionally call spatialNavigation.handlePlayerBlur()  from handleBlur() to store the last selected component on blur.

Reply:
@CarlosVillasenor, @bzizmo 
Do we need to listen to blur/focus event on each component?
Can we move this logic related to handlePlayerFocus and handlePlayerBlur to the spatialNavigation class and listen to these events on the player itself?

eg:

player.on('focus', this.handlePlayerFocus);
player.on('blur', this.handlePlayerBlur);

Reply:
@dzianis-dashkevich both listeners were moved to spatialnavigation class